src/sidebar/store/modules/selection.js

```diff
@@ -506,6 +524,35 @@ function sortKeys(state) {
   return sortKeysForTab;
 }
 
+/* Selectors that take root state */
+
+/**
+ * Retrieve state needed to calculate the root thread
+ *
+ * @return {ThreadState}
```

Comment:
`@return` has no effect unless attached to a function or method declaration AFAIK. I got this to typecheck with the following diff.

```diff
diff --git a/src/sidebar/store/modules/selection.js b/src/sidebar/store/modules/selection.js
index 93df3026b..ba199e886 100644
--- a/src/sidebar/store/modules/selection.js
+++ b/src/sidebar/store/modules/selection.js
@@ -529,16 +529,17 @@ function sortKeys(state) {
 /**
  * Retrieve state needed to calculate the root thread
  *
- * @return {ThreadState}
+ * @type {(rootState: any) => ThreadState}
  */
 const threadState = createSelector(
   rootState => rootState.annotations.annotations,
   rootState => rootState.route.name,
   rootState => rootState.selection,
   (annotations, routeName, selection) => {
-    const filters = {};
-    if (focusModeActive(selection)) {
-      filters.user = focusModeUserFilter(selection);
+    const filters = /** @type {Object.<string,string>} */ ({});
+    const userFilter = focusModeUserFilter(selection);
+    if (userFilter) {
+      filters.user = userFilter;
     }
     const selectionState = {
       expanded: expandedMap(selection),
```

The change to `filters.user` was because `focusModeUserFilter` can return `null`. The previous code worked because `focusModeUserFilter` _currently_ always returns a string when `focusModeActive` returns true, but this connection wasn't obvious to TS. A cast inside the `if (focusModeActive(..))` block would also have worked.